core/src/Config.purs

```diff
@@ -57,17 +56,18 @@ import Registry.Range as Range
 import Registry.Sha256 as Sha256
 import Registry.Version as Version
 import Spago.FS as FS
+import Type.Proxy (Proxy(..))
 
 type Config =
   { package :: Maybe PackageConfig
   , workspace :: Maybe WorkspaceConfig
   }
 
 configCodec :: JsonCodec Config
-configCodec = CAR.object "Config"
-  { package: CAR.optional packageConfigCodec
-  , workspace: CAR.optional workspaceConfigCodec
-  }
+configCodec = CA.object "Config"
+  $ CA.recordPropOptional (Proxy :: _ "package") packageConfigCodec
+  $ CA.recordPropOptional (Proxy :: _ "workspace") workspaceConfigCodec
+  $ CA.record
```

Comment:
I generally stuck to the order I saw defined in the type synonyms. But if you want a different order for any fields just lmk and I can change it.